library/Notifications/Widget/ItemList/EscalationConditionListItem.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/* Icinga Notifications Web | (c) 2023 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Notifications\Widget\ItemList;
+
+use ipl\Html\BaseHtmlElement;
+use ipl\Html\FormElement\BaseFormElement;
+use ipl\Html\FormElement\SubmitButtonElement;
+
+class EscalationConditionListItem extends BaseHtmlElement
+{
+    protected $defaultAttributes = ['class' => 'option'];
+
+    protected $tag = 'li';
+
+    /** @var ?SubmitButtonElement Remove button for the recipient */
+    public $removeButton;
+
+    /** @var BaseFormElement Condition type */
+    public $conditionType;
+
+    /** @var BaseFormElement Operator used for the condition */
+    public $operator;
+
+    /** @var BaseFormElement Condition value */
+    public $conditionVal;
+
+    public function __construct(
+        BaseFormElement $conditionType,
+        BaseFormElement $operator,
+        BaseFormElement $conditionVal,
+        ?SubmitButtonElement $removeButton
+    ) {
+        $this->conditionType = $conditionType;
+        $this->operator = $operator;
+        $this->conditionVal = $conditionVal;
+        $this->removeButton = $removeButton;
+    }
+
+    protected function assemble(): void
+    {
+        $this->add([
+            $this->conditionType,
+            $this->operator,
+            $this->conditionVal,
+            $this->removeButton
+        ]);
```

Comment:
You can call this method directly in constructor. And remove the assemble method.

Reply:
There is a reason, this is done in assemble. While in case of single condition you would have to not show the remove button based on the condition count and whether there is an escalation with no condition. We want the condition list to be assembled as late as possible.